File: dataset/CoT - Copy/GEMINI_gemini-2.5-pro/Task10.c
#include <stdio.h>

/**
 * @brief Given two integers a and b, which can be positive or negative,
 * find the sum of all the integers between and including them and return it.
 * If the two numbers are equal return a or b.
 * 
 * @param a An integer.
 * @param b An integer.
 * @return The sum of all integers between a and b (inclusive).
 */
long long getSum(int a, int b) {
    // If the numbers are the same, return one of them.
    if (a == b) {
        return a;
    }

    // Use long long to prevent potential integer overflow during calculation.
    long long min_val = (a < b) ? a : b;
    long long max_val = (a > b) ? a : b;
    
    // Apply the formula for the sum of an arithmetic series:
    // Sum = n/2 * (first + last)
    // where n is the number of terms (max - min + 1)
    return (max_val - min_val + 1) * (min_val + max_val) / 2;
}

int main() {
    // Test cases
    printf("%lld\n", getSum(1, 0));    // Expected: 1
    printf("%lld\n", getSum(1, 2));    // Expected: 3
    printf("%lld\n", getSum(1, 1));    // Expected: 1
    printf("%lld\n", getSum(-1, 0));   // Expected: -1
    printf("%lld\n", getSum(-1, 2));   // Expected: 2
    return 0;
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task10.java

public class Task10 {
    public static int getSum(int a, int b) {
        if (a == b) {
            return a;
        }
        
        int min = Math.min(a, b);
        int max = Math.max(a, b);
        
        // Using arithmetic progression formula: sum = n * (first + last) / 2
        int n = max - min + 1;
        return n * (min + max) / 2;
    }
    
    public static void main(String[] args) {
        // Test case 1
        System.out.println("Test 1 (1, 0): " + getSum(1, 0) + " (Expected: 1)");
        
        // Test case 2
        System.out.println("Test 2 (1, 2): " + getSum(1, 2) + " (Expected: 3)");
        
        // Test case 3
        System.out.println("Test 3 (0, 1): " + getSum(0, 1) + " (Expected: 1)");
        
        // Test case 4
        System.out.println("Test 4 (1, 1): " + getSum(1, 1) + " (Expected: 1)");
        
        // Test case 5
        System.out.println("Test 5 (-1, 2): " + getSum(-1, 2) + " (Expected: 2)");
    }
}
